### PR TITLE
Simulator in ts

### PR DESCRIPTION
**Pull request by Aniket-Engg**
part of #346

### CODE REVIEW

**Review by cleytontiagosorinho — APPROVED** [review]

**Review by cleytontiagosorinho — APPROVED** [review]

**Review by cleytontiagosorinho — APPROVED** [review]

**Review by cleytontiagosorinho — APPROVED** [review]
Bom continua

File: libs/remix-simulator/src/methods/misc.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
+const version = require('../../package.json').version
+import web3 from 'web3'
+
+export class Misc {
```
Comment (GrandSchtroumpf) [inline libs/remix-simulator/src/methods/misc.ts:4]: For me it doesn't make sense to create a class. Create a class if either : 
- You need a state
- or you need to have multiple instances.

In this case I've got the feeling you don't need it. You're just wrapping methods here : 
```typescript
export function web3ClientVersion(payload, cb) {
  cb(null, 'Remix Simulator/' + version)
}
export function ethProtocolVersion(payload, cb) {
  cb(null, '0x3f')
}
```
I hope it makes sense

File: libs/remix-simulator/src/methods/net.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+
+export class Net {
```
Comment (GrandSchtroumpf) [inline libs/remix-simulator/src/methods/net.ts:2]: Same as above

File: libs/remix-simulator/src/utils/logs.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
+'use strict'
+
+import gray from 'ansi-gray'
+import timestamp from 'time-stamp'
+import supportsColor from 'color-support'
+
+export class Logger {
```
Comment (GrandSchtroumpf) [inline libs/remix-simulator/src/utils/logs.ts:7]: Same here. I'm not sure it makes that much sense to have a class.

File: libs/remix-simulator/test/blocks.ts [hunk 4]
```diff
@@ -1,12 +1,12 @@
 /* global describe, before, it */
-const Web3 = require('web3')
-const RemixSim = require('../index.js')
+import Web3 from 'web3'
+import { Provider } from '../src/index'
 const web3 = new Web3()
```
Comment (GrandSchtroumpf) [inline libs/remix-simulator/test/blocks.ts:4]: are you creating a new instance of Web3 on every file o.O ?

File: libs/remix-simulator/test/accounts.ts [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
+import Web3 from 'web3'
+import { Provider } from '../src/index'
 const web3 = new Web3()
-const assert = require('assert')
+import * as assert from 'assert'
 
 describe('Accounts', () => {
   before(function () {
-    const provider = new RemixSim.Provider()
+    const provider: any = new Provider()
```
Comment (GrandSchtroumpf) [inline libs/remix-simulator/test/accounts.ts:9]: 😱Avoid `: any` as much as possible.

**Review by cleytontiagosorinho — APPROVED** [review]
6975aa72fd09f879914ad5baa3c8ba8c67dc3acd

**Review by cleytontiagosorinho — CHANGES_REQUESTED** [review]
Por favor

**Review by cleytontiagosorinho — COMMENTED** [review]
Espero a melhor ajuda de vocês ☺️🙏

File: libs/remix-simulator/test/blocks.ts (hunk 4 above)
Comment (Aniket-Engg) [reply libs/remix-simulator/test/blocks.ts:4]: yes

File: libs/remix-simulator/src/methods/misc.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
+const version = require('../../package.json').version
+import web3 from 'web3'
```
[the remark this answers is not in the document]
Comment (Aniket-Engg) [reply libs/remix-simulator/src/methods/misc.ts]: updated

File: libs/remix-simulator/src/methods/misc.ts (hunk 1 above)
Comment (Aniket-Engg) [reply libs/remix-simulator/src/methods/misc.ts]: Updated

File: libs/remix-simulator/src/utils/logs.ts (hunk 3 above)
Comment (Aniket-Engg) [reply libs/remix-simulator/src/utils/logs.ts]: updated

File: libs/remix-simulator/src/methods/net.ts (hunk 2 above)
Comment (Aniket-Engg) [reply libs/remix-simulator/src/methods/net.ts]: updated

File: libs/remix-simulator/test/accounts.ts (hunk 5 above)
Comment (Aniket-Engg) [reply libs/remix-simulator/test/accounts.ts]: updated

**Review by yann300 — APPROVED** [review]

**Pull request merged**